src/Umbraco.Web/Extensions/UdiExtensions.cs

```diff
@@ -36,8 +36,30 @@ public static IPublishedContent ToPublishedContent(this Udi udi)
                         return umbracoHelper.TypedMember(memberAttempt.Result);
                     break;
             }
-            
+
             return null;
         }
+
+        /// <summary>
+        /// An extension method to easily acquire the integer Id for a given udi
+        /// </summary>
+        /// <param name="udi"></param>
+        /// <returns>An <see cref="int"/> identifier if the item is found, -1 otherwise</returns>
+        public static int ToIntId(this Udi udi)
```

Comment:
This method is just as bad as the other `ToPublishedContent` extension method and still uses hidden singletons and isn't testable, this also performs an entire object lookup which isn't necessary. 

There is no need to ToString a UDI just to parse it back out, the UDI is already a UDI! You should just cast Udi to GuidUdi and if it works then you already have access to the GUID. If it doesn't cast than it's a StringUdi for which you can throw a NotSupportedException

I would suggest just adding a method to the `IEntityService`. We already have `GetIdForKey` and `GetKeyForId` so add: `GetIdForUdi` 

Or you could just add `GetIdForUdi` as an extension method to the `IEntityService`, then you aren't relying on singletons, etc... However, extensions methods are not mockable so I would still just go ahead and add `GetIdForUdi` to the `IEntityService`